pkg/curatedpackages/bundle.go

```diff
@@ -0,0 +1,100 @@
+package curatedpackages
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+
+	api "github.com/aws/eks-anywhere-packages/api/v1alpha1"
+	"github.com/aws/eks-anywhere-packages/pkg/artifacts"
+	"github.com/aws/eks-anywhere-packages/pkg/bundle"
+	"github.com/aws/eks-anywhere-packages/pkg/testutil"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/dependencies"
+	"github.com/aws/eks-anywhere/pkg/executables"
+)
+
+const (
+	RegistryBaseRef = "public.ecr.aws/q0f6t3x4/eksa-package-bundles"
+)
+
+func GetLatestBundle(ctx context.Context, kubeConfig string, source BundleSource, kubeVersion string) (*api.PackageBundle, error) {
+	switch source {
+	case Cluster:
+		return getActiveBundleFromCluster(ctx, kubeConfig)
+	case Registry:
+		return getLatestBundleFromRegistry(ctx, kubeVersion)
+	default:
+		return nil, fmt.Errorf("unknown source: %q", source)
+	}
+}
+
+func createBundleManager(kubeVersion string) bundle.Manager {
+	versionSplit := strings.Split(kubeVersion, ".")
+	major, minor := versionSplit[0], versionSplit[1]
+	log := logr.Discard()
+	discovery := testutil.NewFakeDiscovery(major, minor)
+	puller := artifacts.NewRegistryPuller()
+	return bundle.NewBundleManager(log, discovery, puller)
+}
+
+func getLatestBundleFromRegistry(ctx context.Context, kubeVersion string) (*api.PackageBundle, error) {
+	bm := createBundleManager(kubeVersion)
+	return bm.LatestBundle(ctx, RegistryBaseRef)
+}
+
+func getActiveBundleFromCluster(ctx context.Context, kubeConfig string) (*api.PackageBundle, error) {
+	deps, err := newDependencies(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("unable to initialize executables: %v", err)
+	}
+	kubectl := deps.Kubectl
+
+	// Active Bundle is set at the bundle Controller
+	bundleController, err := getActiveController(ctx, kubectl, kubeConfig)
+	if err != nil {
+		return nil, err
+	}
+	bundle, err := getPackageBundle(ctx, kubectl, kubeConfig, bundleController.Spec.ActiveBundle)
+	if err != nil {
+		return nil, err
+	}
+	return bundle, nil
+}
+
+func getPackageBundle(ctx context.Context, kubectl *executables.Kubectl, kubeConfig string, activeBundle string) (*api.PackageBundle, error) {
```

Comment:
We might want to create a `CuratedPackagesKubeClient` for these two methods, just wrapping around the `kubectl` executable

Reply:
Up above there was talk about making BundleSource an interface and now moving some of this code to a new executable. I'd chose one or the other and I think the derived BundleSource idea was a bit better.